### PR TITLE
feat(tariscript)!: use varints in tari script (de)serialization

### PR DESCRIPTION
**Pull request by Cifko**
Description
---
Use varint for op codes that use 64bit numbers but in general will be low.

Motivation and Context
---
https://github.com/tari-project/tari/issues/4108

How Has This Been Tested?
---
Unit tests.

### CODE REVIEW

File: infrastructure/tari_script/src/op_codes.rs [hunk 1]
```diff
@@ -623,18 +621,15 @@ mod test {
...
-            let s = &[val, 63, 0, 0, 0, 0, 0, 0, 0, 1, 2, 3];
+            assert!(matches!(Opcode::read_next(&[val, 255]), Err(ScriptError::InvalidData)));
+            let s = &[val, 63, 1, 2, 3];
             let (opcode, rem) = Opcode::read_next(s).unwrap();
             assert_eq!(opcode, *op);
             assert_eq!(rem, &[1, 2, 3]);
             // Deserialise
             let mut arr = vec![1, 2, 3];
             op.to_bytes(&mut arr);
-            assert_eq!(&arr, &[1, 2, 3, val, 63, 0, 0, 0, 0, 0, 0, 0]);
```
Comment (stringhandler) [inline infrastructure/tari_script/src/op_codes.rs:637]: This is weird that this passes. The op code is in the middle of the value. I was expecting 63 to occur before the value

**Review by stringhandler — CHANGES_REQUESTED** [review]

**Review by sdbondi — DISMISSED** [review]

File: infrastructure/tari_script/src/op_codes.rs (hunk 1 above)
Comment (stringhandler) [reply infrastructure/tari_script/src/op_codes.rs:637]: my bad, 63 is the height

**Review by stringhandler — DISMISSED** [review]

**Comment by stringhandler** [comment]
Just needs clippy to pass and can merge

File: infrastructure/tari_script/src/op_codes.rs (hunk 1 above)
Comment (sdbondi) [reply infrastructure/tari_script/src/op_codes.rs:637]: The existing test is confusing, 63 is the height value and `val` is the op code.
Should use the constants in tests and pass in a `val` and `expected_opcode` into `test_check_height`

**Review by sdbondi — APPROVED** [review]

**Pull request merged**